### DISCUSSION TITLE
How to add an icon to Circular Progress component?

### DISCUSSION THREAD

**Discussion by Vincent1334** [zauberzeug/nicegui#1491] [Q&A]
### Question

Hey,
how is it possible to display an icon next to the Value label?
I am trying to modify the example.

```python
from nicegui import ui

slider = ui.slider(min=0, max=1, step=0.01, value=0.5)
ui.circular_progress().bind_value_from(slider, 'value')

ui.run()
```

**Comment by rodja** [comment]
In NiceGUI you can put elements inside each other by using the `with` statement. The logical solution therefore would be 

```py
slider = ui.slider(min=0, max=1, step=0.01, value=0.5)
with ui.circular_progress(show_value=True).bind_value_from(slider, 'value'):
    ui.icon('ring_volume').classes('text-2xl')
```

But the label and icon will then overlap. There is a parameter `show_value=False` to hide the label. But that is also disabling the whole nesting capability of the internally used `q-circular-progress`. I'll create a pull request to improve on that in later versions. Right now you can make it work with

```py
slider = ui.slider(min=0, max=1, step=0.01, value=0.5)
with ui.circular_progress(show_value=True).bind_value_from(slider, 'value') as progress:
    progress.clear()
    ui.icon('ring_volume').classes('text-2xl')
```

**Comment by pythonormen** [comment]
I have something like this in my circular progress. I think I also tried to add slot the quasar way but don't think I got it to work.

```python
from nicegui import ui

slider = ui.slider(min=0, max=1, step=0.01, value=0.5)
with ui.circular_progress(size='60px').bind_value_from(slider, 'value') as progress:
    with progress.add_slot('default'):
        ui.icon('ring_volume')
        ui.label().bind_text_from(progress, 'value')

ui.run()
```

**Comment by rodja** [comment]
I just created pull request https://github.com/zauberzeug/nicegui/pull/1493 which simplifies putting elements inside a circular progress.
